src/devices/acurite.c

```diff
@@ -729,7 +729,7 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
             // 11 bits needed for specified range -40 C to 70 C (-40 F - 158 F)
             // range -100 C to 1538.4 C
             int temp_raw = ((bb[4] & 0x7F) << 7) | (bb[5] & 0x7F);
-            tempc = temp_raw * 0.1 - 100;
+            tempc = temp_raw * 0.1f - 100;
```

Comment:
there is still a double lurking here. prefered would be `(temp_raw - 1000) * 0.1f`

Reply:
or maybe it's an implicit float…

Reply:
According to VC++, it gets indeed converted to a `float` not a `double`, there is no longer a warning on that line.

Reply:
Note that I hesitated changing `100` to `100.0f` but I'm not sure it makes it any clearer...

Reply:
temp_raw should be converted to float and then -100 should be converted to float and then everything is float

Reply:
my point was an integer substraction, then a float multiply instead of two float ops.

Reply:
The rules give that it should be float. But I see your point if the int is large.

Reply:
The initial goal of that PR was to address the warnings by simply adding the suffix wherever required. 
That being said, I agree that there are optimization opportunities at various places (`value / 3600.0f / 1000.0f` comes to mind) but shouldn't this be addressed in another PR?

Reply:
True, and greping for `f - ` afterwards will find most of those instances.